src/Bicep.Core/Parser/Parser.cs

```diff
@@ -57,45 +57,81 @@ public ProgramSyntax Program()
         }
 
         public SyntaxBase Declaration() =>
-            this.WithRecovery(() =>
-            {
-                var current = reader.Peek();
+            this.WithRecovery(
```

Comment:
Now that declaration syntax no longer includes the newLine token, is there actually a good reason for the "consumeTerminator" flag to still exist? If it can be replaced with slightly more verbose parsing logic, I'd be in favor of replacing it, as it makes the parsing code harder to reason about since you need to carefully consider which flags are being passed to `WithRecovery` when reading the code.

Reply:
We are consuming the terminator in one place (parsing the newline that is required after a declaration).

Reply:
Leaving as-is